src/degeneracy.jl

```diff
@@ -39,35 +37,50 @@ julia> core_number(g)
 """
 function core_number(g::AbstractGraph{T}) where T
     has_self_loops(g) && throw(ArgumentError("graph must not have self-loops"))
-    degrees = degree(g)
-    vs = sortperm(degrees)
-    bin_boundaries = [1]
-    curr_degree = 0
-    for (i, v) in enumerate(vs)
-        if degrees[v] > curr_degree
-            append!(bin_boundaries, repeat([i], (degrees[v] - curr_degree)))
-            curr_degree = degrees[v]
-        end
+    n = nv(g)
+    deg = T.(degree(g)) # degree should really return T.
+    maxdeg = maximum(deg)
+    bin = zeros(T, maxdeg+1)
```

Comment:
This makes a lot of very short names here, should we either comment on what each is and/or use some longer names for the ones used throughout the function (not the ones local to a single for loop or so

Reply:
I'll defer to @abhinavmehndiratta to provide better variable names.